sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -154,12 +154,19 @@ export class CallArgSerializer {
     return hasTxContext ? params.slice(0, params.length - 1) : params;
   }
 
-  async newObjectArg(objectId: string): Promise<ObjectArg> {
+  async newObjectArg(
+    objectId: string,
+    expectedType: SuiMoveNormalizedType,
+  ): Promise<ObjectArg> {
     const object = await this.provider.getObject(objectId);
     const initialSharedVersion = getSharedObjectInitialVersion(object);
 
-    const mutable = true; // Defaulted to True to match current behavior.
     if (initialSharedVersion) {
+      const mutable =
+        typeof expectedType === 'object' && 'MutableReference' in expectedType
```

Comment:
@healthydeve do you mind checking if we have done something similar in the RPC transaction builder on the Rust side for this? @andll is this needed?

Reply:
At least in the case of `Clock`, yes we will need to do something similar on the Rust side as well.

Reply:
Added to the rust sdk side.